File: easy/3498-reverse-degree-of-a-string/solution.py
class Solution:
    def reverseDegree(self, s: str) -> int:
        """
        3498. Reverse Degree of a String
        Time: O(n)
        Space: O(1)
        """
        total = 0
        for i, ch in enumerate(s):
            weight = 26 - (ord(ch) - ord("a"))
            total += weight * (i + 1)
        return total

File: easy/3498-reverse-degree-of-a-string/solution.java
/**
 * 3498. Reverse Degree of a String
 * Time: O(n)
 * Space: O(1)
 */
class Solution {
    public int reverseDegree(String s) {
        int total = 0;
        for (int i = 0; i < s.length(); i++) {
            int weight = 26 - (s.charAt(i) - 'a');
            total += weight * (i + 1);
        }
        return total;
    }
}
